QueryEngine/ResultSetIteration.cpp

```diff
@@ -2173,6 +2173,50 @@ bool ResultSetStorage::isEmptyEntryColumnar(const size_t entry_idx,
   return false;
 }
 
+namespace {
+
+template <typename T>
+inline size_t makeBinSearch(size_t l, size_t r, T&& is_empty_fn) {
+  // Avoid search if there are no empty keys.
+  if (!is_empty_fn(r - 1)) {
+    return r;
+  }
+
+  --r;
+  while (l != r) {
+    size_t c = (l + r) / 2;
+    if (is_empty_fn(c)) {
+      r = c;
+    } else {
+      l = c + 1;
+    }
+  }
+
+  return r;
+}
+
+}  // namespace
+
+size_t ResultSetStorage::binSearchRowCount() const {
+  CHECK(query_mem_desc_.getQueryDescriptionType() == QueryDescriptionType::Projection);
+  CHECK_EQ(query_mem_desc_.getEffectiveKeyWidth(), 8);
```

Comment:
Can you please change this to be using `size_t(8)` instead? It currently produces a warning because of type mismatch in comparison.